generators/chipyard/src/main/scala/example/GCD.scala

```diff
@@ -233,3 +332,8 @@ class WithGCD(useAXI4: Boolean = false, useBlackBox: Boolean = false) extends Co
   case GCDKey => Some(GCDParams(useAXI4 = useAXI4, useBlackBox = useBlackBox))
 })
 // DOC include end: GCD config fragment
+
+// useHLS cannot be used with useAXI4 and useBlackBox
+class WithHLSGCD extends Config((site, here, up) => {
+  case GCDKey => Some(GCDParams(useAXI4 = false, useBlackBox = false, useHLS = true))
+})
```

Comment:
Can you just reuse the above fragment and have a check to ensure only one flag is enabled at a time.